docs/Contributing/API-for-contributors.md

```diff
@@ -1942,18 +1942,20 @@ If the `name` is not already associated with an existing team, this API route cr
+| software.app_store_apps.self_service           | object   | body  | If null, turn off self-service for this app. Otherwise, use the object properties for scoping. Only one of the three object fields must be set. |
+| app_store_apps.self_service.all_hosts     | boolean   | body  | If set, allow all compatible hosts on the team to use self-service for this app. |
+| app_store_apps.self_service.labels_include_any     | array   | body  | Allow only compatible hosts with one or more of the supplied tags to install the app via self-service. |
+| software.packages.self_service.labels_exclude_any     | array   | body  | Allow only compatible hosts with none of the supplied tags to install the app via self-service. |
 | software.packages                          | array   | body  | An array of objects with values below. |
 | software.packages.url                      | string   | body  | URL to the software package (PKG, MSI, EXE or DEB). |
 | software.packages.install_script           | string   | body  | Command that Fleet runs to install software. |
 | software.packages.pre_install_query        | string   | body  | Condition query that determines if the install will proceed. |
 | software.packages.post_install_script      | string   | body  | Script that runs after software install. |
 | software.packages.uninstall_script       | string   | body  | Command that Fleet runs to uninstall software. |
-| software.packages.self_service           | boolean   | body  | If `true` lists software in the self-service. |
-| software.packages.labels_include_any     | array   | body  | Target hosts that have any label in the array. Only one of `labels_include_any` or `labels_exclude_any` can be included. If neither are included, all hosts are targeted. |
-| software.packages.labels_exclude_any     | array   | body  | Target hosts that don't have any label in the array. Only one of `labels_include_any` or `labels_exclude_any` can be included. If neither are included, all hosts are targeted. |
+| software.packages.self_service           | object   | body  | If null, turn off self-service for this package. Otherwise, use the object properties for scoping. Only one of the three object fields must be set. |
+| software.packages.self_service.all_hosts     | boolean   | body  | If set, allow all compatible hosts on the team to use self-service for this package. |
+| software.packages.self_service.labels_include_any     | array   | body  | Allow only compatible hosts with one or more of the supplied tags to install the package via self-service. |
+| software.packages.self_service.labels_exclude_any     | array   | body  | Allow only compatible hosts with none of the supplied tags to install the package via self-service. |
```

Comment:
Have you considered having this be `software.packages.self-service.scope`, which accepts either `all_hosts`, `labels_include_any`, or `labels_exclude_any`? I feel like that's a little simpler to understand as a user. (I realize this is a contributor endpoint though.)

Reply:
Not quite following. Are you suggesting taking the same structure (3 keys, one of which is a boolean, the other two of which are string[]) and nesting it one layer deeper, such that `self_service` just has one key, or something else?

The current revision mirrors the `self_service` key on the response JSON elsewhere. albeit with strings rather than full objects for labels because labels are referred to by name in GitOps endpoints as we aren't guaranteed to have ID yet.

Reply:
@iansltx Ah wait never mind, I was thinking "why have 3 separate things that you can only use 1 of?" but the `labels_…` are arrays.  Ignore this! (I think I got stuck on the new `all_hosts` key; this tells me I should probably not save big blocks of API design reviews for Friday afternoon. 😅)